web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -33,17 +39,81 @@ export async function sendAndConfirmTransaction(
     sendOptions,
   );
 
-  const status = (
-    await connection.confirmTransaction(
-      signature,
-      options && options.commitment,
-    )
-  ).value;
+  const subscriptionCommitment =
+    options?.preflightCommitment || options?.commitment;
+  let subscriptionId: number | undefined;
+  let status: RpcResponseAndContext<SignatureResult> | null = null;
+  let message: string | null = null;
```

Comment:
Since you're returning your status code from your promises now, you don't need this.